src/main/java/com/csc530/sat/type/DDType.java

```diff
@@ -0,0 +1,15 @@
+package com.csc530.sat.type;
+
+/**
+ * A type that the decision diagram can reason about
+ *
+ * @param <T>
+ *            the java type that this type describes
+ */
+public interface DDType<T> {
```

Comment:
This isn't a type, this is a wrapper class around a value T. Why not reason about T?